php/php.editor/src/org/netbeans/modules/php/editor/verification/ReturnStatementWithoutValueHintError.java

```diff
@@ -0,0 +1,218 @@
+        if (phpParseResult.getProgram() == null) {
+            return;
+        }
+        FileScope fileScope = context.fileScope;
+        fileObject = phpParseResult.getSnapshot().getSource().getFileObject();
+        if (fileScope != null && fileObject != null && appliesTo(fileObject)) {
+            if (CancelSupport.getDefault().isCancelled()) {
+                return;
+            }
+            CheckVisitor checkVisitor = new CheckVisitor();
+            phpParseResult.getProgram().accept(checkVisitor);
+            Map<ASTNode, Set<ReturnStatement>> returnStatements = checkVisitor.getReturnStatements();
+            checkReturnType(returnStatements, hints);
+        }
+    }
+
+    protected PhpVersion getPhpVersion(@NullAllowed FileObject file) {
+        if (file == null) {
+            return PhpVersion.getDefault();
+        }
+        return CodeUtils.getPhpVersion(file);
+    }
+
+    private boolean appliesTo(FileObject file) {
+        return getPhpVersion(file).compareTo(PhpVersion.PHP_70) >= 0;
+    }
+
+    private void checkReturnType(Map<ASTNode, Set<ReturnStatement>> returnStatements, List<Hint> hints) {
+        for (Entry<ASTNode, Set<ReturnStatement>> entry : returnStatements.entrySet()) {
+            if (CancelSupport.getDefault().isCancelled()) {
+                return;
+            }
+            ASTNode node = entry.getKey();
+            Expression returnType = null;
+            if (node instanceof FunctionDeclaration) {
+                returnType = ((FunctionDeclaration) node).getReturnType();
+            } else if (node instanceof LambdaFunctionDeclaration) {
+                returnType = ((LambdaFunctionDeclaration) node).getReturnType();
+            }
+
+            if (returnType == null) {
+                continue;
+            }
+            Set<ReturnStatement> statements = entry.getValue();
+
+            if (returnType instanceof NamespaceName) {
+                NamespaceName namespaceName = (NamespaceName) returnType;
+                String name = CodeUtils.extractUnqualifiedName(namespaceName);
+                checkReturnStatementsWithoutValue(statements, name, hints);
+            } else if (returnType instanceof NullableType
+                    || returnType instanceof UnionType
+                    || returnType instanceof IntersectionType) {
+                checkReturnStatementsWithoutValue(statements, "", hints);
+            }
+
+        }
+    }
+
+    @NbBundle.Messages({
+        "ReturnStatementWithoutValueHintErrorDesc=Return statement must be filled with the value"
+    })
+    private void checkReturnStatementsWithoutValue(Set<ReturnStatement> statements, String name, List<Hint> hints) {
+        if (!Type.VOID.equals(name) && !isNeverType(name)) {
+            // check empty return statement
+            statements.forEach((statement) -> {
+                if (CancelSupport.getDefault().isCancelled()) {
+                    return;
```

Comment:
this will return from lambda, not from checkReturnStatementsWithoutValue
is it intentional?

Reply:
I got this code from here [php/php.editor/src/org/netbeans/modules/php/editor/verification/ReturnTypeHintError.java](https://github.com/apache/netbeans/blob/cef11a8c61656b7e15562a9d4c28e9d386a1d7f1/php/php.editor/src/org/netbeans/modules/php/editor/verification/ReturnTypeHintError.java#L135-L148).
Is it wrong?

Reply:
I don't know. I just spotted a strange costruct and brought it to your attention. Maybe it is perfectly fine.

Reply:
good catch. This looks like a little cancelability bug. Both cases should be reverted to normal loops. This is probably from the time when NB had a hint which recommended to use the for each loop - and someone did this by accident and/or blindly. I turned that hint off via #3300.

If you want you can simply add another commit to this PR which fixes this and we merge this unsquashed. Might be worth checking if this happens in other hints too.

Reply:
this isn't bad performance wise since it would just loop without doing anything when canceled. But this can introduce bugs later e.g if someone adds something after the `foreach` loop.

Reply:
on second thought, this is so minor that you can simply add this to your first commit. feel free to force push if you find other occurrences :)

Reply:
Fixed.